libr/asm/p/asm_tricore.c

```diff
@@ -53,7 +53,24 @@ static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 	disasm_obj.fprintf_func = &generic_fprintf_func;
 	disasm_obj.stream = stdout;
 
-	disasm_obj.mach = 2; // select CPU TYPE
+	// cpu type
+	if (a->cpu && *a->cpu) {
+		if (!strcmp (a->cpu, "generic")) {
+			disasm_obj.mach = 0x00000000;
```

Comment:
What about using enum?

Reply:
Yeah i was thinking that too. But then I was not able to come up with a solution. Do you have  one?

Reply:
Just create enum:
```c
enum {
   TRICORE_CPU_GENERIC = 0x000000,
   TRICORE_CPU_RIDER_A = 0x000001,
   ...
}
```

Reply:
Oh you mean when assigning value use enum, i thought to use enum in comparison logic.

Reply:
maybe worth moving this switch case into a separate function like disasm_obj.mach = cpu_to_mach (a->cpu);

Reply:
okay